tests/infra/ccf.py

```diff
@@ -275,11 +296,44 @@ def start_and_join(self, args):
         self.status = ServiceStatus.OPEN
         LOG.success("***** Network is now open *****")
 
-    def start_in_recovery(self, args, ledger_file):
-        self.common_dir = get_common_folder_name(args.workspace, args.label)
+    def start_in_recovery(self, args, ledger_file, common_dir=None):
+        self.common_dir = common_dir or get_common_folder_name(
+            args.workspace, args.label
+        )
+        self.store_current_network_encryption_key()
```

Comment:
Thinking about this more, I want to make it more explicit that members have to use the network encryption public key of the previous network in order to decrypt the recovery shares. I will pass it as an extra recovery argument to `start_test_network.sh`, along with the ledger and the common directory.

Reply:
Done.